ykcompile/src/lib.rs

```diff
@@ -236,14 +257,12 @@ impl<TT> TraceCompiler<TT> {
             let tyid = self.local_decls[&l].ty;
             if Self::can_live_in_register(&tyid) {
                 // Find a free register to store this local.
-                let loc = if let Some(reg) = self.register_content_map.iter().find_map(|(k, v)| {
-                    if v == &None {
-                        Some(*k)
-                    } else {
-                        None
-                    }
-                }) {
-                    self.register_content_map.insert(reg, Some(l));
+                let loc = if let Some(reg) =
```

Comment:
I wonder if we should factor this out into a method now? It'd make it easier to decouple the allocator later?

Reply:
I see this was done in a later commit, so ignore this.